li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/largemessage/ConsumerRecordsProcessor.java

```diff
@@ -421,7 +423,12 @@ public long getRecordsSkipped() {
     K key = _keyDeserializer.deserialize(tp.topic(), consumerRecord.key());
     byte[] valueBytes = parseAndMaybeTrackRecord(tp, consumerRecord.offset(), consumerRecord.value());
     if (valueBytes != INCOMPLETE_RESULT) {
-      V value = _valueDeserializer.deserialize(tp.topic(), valueBytes);
+      V value;
```

Comment:
perf nit - instead of performing an instanceof per record, can we do this once and use a delegate here to call the "right" method?